src/csharp/Microsoft.Spark/Microsoft.Spark.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;netcoreapp2.1;netstandard2.1</TargetFrameworks>
```

Comment:
@eerhardt Should I add `netcoreapp3.1` instead of replacing `netcoreapp2.1`?

Reply:
For libraries (like `Microsoft.Spark.csproj`), you don't need to target all the frameworks you support running on. You just need to target the lowest/earliest frameworks that you support.

For example, if you just target `netstandard2.0`, your library will run on any desktop .NET Framework 4.6.1 or higher. And it will run on .NET Core 2.0 or higher.

The reason you would multi-target a library is so you can have different code that will execute when running on different frameworks. For example, .NET Core 3.1 adds a new API you want to use, so you multi-target to `netstandard2.0;netcoreapp3.1`. And now you can use `#if` conditions in your code to compile different code when building for `netstandard2.0`, and use the new API when building for `netcoreapp3.1`.

Since you don't have `netcoreapp3.1` specific code in this library, it isn't necessary to put `netcoreapp3.1` in this list at all.

The only framework-specific code I see in this project is:

https://github.com/dotnet/spark/blob/8dc789fa6cc5d60eecb5d787cc04ea3a366e2035/src/csharp/Microsoft.Spark/Sql/Types/DataType.cs#L208-L219

So in all actuality, if you want to drop 1st class support for .NET Core 2.1, you would really only need:

`<TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>` on this line. (Because .NET Core 3.1 is compatible with `netstandard2.1`, so the `netstandard2.1` compiled assemblies will be used when running on .NET Core 3.1.)

Reply:
Thanks @eerhardt. `netcoreapp2.1` was added from this discussion: https://github.com/dotnet/spark/pull/364#discussion_r354627266. I will remove `netcoreapp2.1` for now and will add it if it is absolutely needed.

Reply:
The only affect this will cause is if someone is on .NET Core 2.1 and uses the `Microsoft.Spark` library, the above code will use the slower Substring/ToLower code since the `netstandard2.0` built version will be used. That doesn't seem too bad, especially if you aren't supporting .NET Core 2.1 on the worker anymore.

Reply:
Yea, and this code is driver side and not used by worker anyway. Thanks @eerhardt!